### ISSUE TITLE
Some messages are dropped

### ISSUE THREAD

**Issue by vivien** [glowing-bear/glowing-bear#738]
From within Glowing Bear, random messages don't get sent. For instance, typing the following:

````
k
ok
````

correctly sent "ok" to the chan, but not "k". Here's the corresponding `/relay raw` snippet:

````
101253 [1] weechat --> ? PING(76) input 0x2b94d90 k
101253 [1] weechat --> ? (77) input 0x2b94d90 /buffer set hotlist -1
101253 [1] weechat --> ? (78) input 0x2b94d90 /input set_unread_current_buffer
101257 [1] weechat <-- ? obj: 242/343 bytes (30%, 0ms), id: _buffer_line_added
101258 [1] weechat <-- ? obj: 281/427 bytes (35%, 0ms), id: _buffer_line_added
101258 [1] weechat <-- ? obj: 198/284 bytes (31%, 0ms), id: _nicklist_diff
101259 [1] weechat <-- ? obj: 245/342 bytes (29%, 0ms), id: _buffer_line_added
101300 [1] weechat --> ? (79) input 0x2b94d90 ok
101300 [1] weechat --> ? (80) input 0x2b94d90 /buffer set hotlist -1
````

The only difference I spot is the "PING" prefix for the "k" input. I cannot tell yet if it is a Glowing Bear issue or Weechat not correctly handling some messages.

Not that this bug does not seem to happen from within WeeChat.
The bug happens on both glowing-bear.org and latest.glowing-bear.org.

Cc'ing: @flashcode

**Comment by vivien** [comment]
That happened again when trying to type "Evropi ^":

````
102901 [1] weechat --> ? PINGPINGPINGPINGPINGPINGPINGPINGPINGPINGPINGPINGPINGPINGPINGPING(85) input 0x2b94d90 Evropi: ^
````

**Comment by flashcode** [comment]
The correct id in a relay message sent to WeeChat is `(76)` but not `PING(76)`. So for me the bug is on glowing-bear side.

Not related note: using pointers in "input" command is not recommended (not safe to use pointers in relay protocol). The complete buffer name should be used instead, for example: `input irc.freenode.#test Hello, world!`

**Comment by lorenzhs** [comment]
What's interesting is that I can't find any calls to `formatPing`, nor the number 76 (neither in decimal nor hex) in the code. What part of GB's code is sending those pings? :O 

Regarding pointers, we actually switched back to using pointers if the WeeChat version is at least 1.0 because we had issues with spaces and other special characters in fullName and WeeChat >= 1.0 validates pointers. I believe this came up with @torhve's matrix script. See https://github.com/glowing-bear/glowing-bear/blob/8b045049ceec2f403f8072653b5828c838c680c8/js/models.js#L483

**Comment by lorenzhs** [comment]
@vivien what browser are you using? 

```
<+xt> and I think the PING is part of the websocket protocol, no?
<+lorenzh> I think so, 'cause we're not sending it
```

**Comment by flashcode** [comment]
Fixed in WeeChat (see weechat/weechat#675).
Feedback is welcome in case of problem (many changes in frame decoding).

**Comment by torhve** [comment]
I have tested this patch and can confirm problem is gone. Thanks @flashcode for the fix.
